src/app/shared/components/xterm-terminal/component.ts

```diff
@@ -0,0 +1,215 @@
+
+  constructor(
+    private readonly _activatedRoute: ActivatedRoute,
+    private readonly _clusterService: ClusterService,
+    private readonly _notificationService: NotificationService,
+    private readonly _renderer: Renderer2,
+    private readonly websocketService: WebsocketService
+  ) {}
+
+  ngOnInit(): void {
+    this.initSubscriptions();
+  }
+
+  initSubscriptions() {
+    this.projectId = this._activatedRoute.snapshot.paramMap.get(PathParam.ProjectID);
+    this.clusterId = this._activatedRoute.snapshot.paramMap.get(PathParam.ClusterID);
+
+    // Note:
+    // Get Cluster Info inorder to show cluster name in shell
+    if (this.projectId && this.clusterId) {
+      this._clusterService
+        .cluster(this.projectId, this.clusterId)
+        .pipe(take(1))
+        .subscribe({
+          next: (cluster: Cluster) => {
+            this.cluster = cluster;
+            this.initTerminal();
+          },
+          error: _ => {},
+        });
+    }
+
+    // Subscription bidirectional communication
+    this.websocketService.messages$.pipe(takeUntil(this._unsubscribe)).subscribe((data: ITerminalFrame) => {
+      this.handleWebSocketConnectionMessages(data);
+    });
+
+    // Subscription connection success
+    this.websocketService
+      .getWebSocketOnConnectObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this._notificationService.success('Successfully connected to cloud shell terminal');
+        this.isConnectionLost = false;
+      });
+
+    // Subscription connection lost
+    this.websocketService
+      .getWebSocketOnCloseObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this.isConnectionLost = true;
+      });
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+    this.websocketService.close();
+
+    // Clean 'resize' Event listener subscription
+    this.removeEventListener();
+  }
+
+  ngAfterViewInit(): void {
+    this.connectToWebSocketConnection();
+  }
+
+  initTerminal(): void {
+    if (this.terminal) {
+      this.terminal.dispose();
+    }
+
+    this.terminal = new Terminal({
+      fontSize: 14,
+      fontFamily: 'Consolas, "Courier New", monospace',
+      bellStyle: 'sound',
+      cursorBlink: true,
+      theme: {
+        foreground: '#f8f8f8',
+        background: '#2b3035',
+      },
+    });
+
+    const containerElement = this.terminalRef.nativeElement;
+    const fitAddon = new FitAddon();
+    this.terminal.loadAddon(fitAddon);
+    this.terminal.open(containerElement);
+
+    const clusterName = this.cluster && this.cluster.name;
+    this.terminal.write('\x1b[37mWEB TERMINAL\x1b\r\n\n');
+    this.terminal.write('\x1b[37mWelcome to Cloud Shell! Type "help" to get started.\r\n\n');
+    this.terminal.write(
+      `\x1b[37mYour Cloud Platform project in this session is set to \x1b[1;34m${clusterName}\x1B[0m\n\n\r`
```

Comment:
@cschieder Should we write this every time? I think it should be really short if we want to add something.